### PR TITLE
Misc fixes: Stokes dataset connectivity, distributed manager initialization and error messages

### PR DESCRIPTION
**Pull request by ktangsali**
# Modulus Pull Request

## Description
This PR fixes a few following issues: 
1. Stokes dataset was not capturing all the edges of a graph. 
2. The error message printed by uninitialized distributed manager was incomplete. 
3. Distributed manager was uninitialized in the load checkpoint utility. 

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/modulus/blob/main/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/modulus/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/modulus/issues) is linked to this pull request.

### CODE REVIEW

**Comment by ktangsali** [comment]
Stokes dataset corrections:

image

Results after `train.py` and `pi_fine_tuning.py`:

image

**Comment by ktangsali** [comment]
Error before and after distributed manager changes:

Before:

image

After:

image

**Review by mnabian — APPROVED** [review]

File: modulus/launch/utils/checkpoint.py [hunk 1]
```diff
@@ -74,6 +74,7 @@ def _get_checkpoint_filename(
     # can save their checkpoint. In the case without model parallelism,
     # model_parallel_rank should be the same as the process rank itself and
     # only rank 0 saves
+    DistributedManager.initialize()
```
Comment (akshaysubr) [inline modulus/launch/utils/checkpoint.py:77]: This isn't the right fix here I think. This will cause other issues if the user already initialized the DistributedManager. Maybe a guard here would be better:
```python
if not DistributedManager.is_initialized():
    logger.warn("`DistributedManager` not initialized already. Initializing now, but this might lead to unexpected errors")
    DistributedManager.initialize()
```

File: modulus/distributed/manager.py [hunk 2]
```diff
@@ -48,11 +48,13 @@ class ModulusUninitializedDistributedManagerWarning(Warning):
     """Warning to indicate usage of an uninitialized DistributedManager"""
 
     def __init__(self):
-        message = "A DistributedManager object is being instantiated before "
-        "this singleton class has been initialized. Instantiating a manager before "
-        "initialization can lead to unexpected results where processes fail "
-        "to communicate. Initialize the distributed manager via "
-        "DistributedManager.initialize() before instantiating."
+        message = (
```
Comment (akshaysubr) [inline modulus/distributed/manager.py:51]: Why are these changes needed?

**Review by akshaysubr — CHANGES_REQUESTED** [review]

File: modulus/distributed/manager.py (hunk 2 above)
Comment (ktangsali) [reply modulus/distributed/manager.py:51]: It wasn't printing the whole error message as shown in the comment here: https://github.com/NVIDIA/modulus/pull/549#issuecomment-2163600705

File: modulus/launch/utils/checkpoint.py (hunk 1 above)
Comment (ktangsali) [reply modulus/launch/utils/checkpoint.py:77]: Good call, let me test this

Comment (ktangsali) [reply modulus/launch/utils/checkpoint.py]: Updated

**Review by akshaysubr — APPROVED** [review]
